types/knex.d.ts

```diff
@@ -875,7 +875,7 @@ declare namespace Knex {
   //
 
   class Client extends events.EventEmitter {
-    constructor(config: Config);
+    constructor(config: Config | string);
     config: Config;
```

Comment:
Not sure if this call should also accept string or not.

Reply:
@enapupe Are you sure about this? I checked the code and I'm pretty sure clients (at least base one and PostgreSQL) are going to explode if you pass string to them, because of e. g. `config.pool.max !== 0`
And `parseConnectionString` is not called anywhere in the code, so I assume what you want to do is to call `parseConnectionString` with string, get instance of Config and pass that to client constructor. And you don't need to modify typings for that.

Reply:
Totally not sure. I have no idea if the `config` call should also allow string. 
I'm positive the `constructor` should. That's why I haven't touched the  referred line. I appreciate you checking it

Reply:
@enapupe config is not a method, it is a field, you don't call it. And please check source code for client constructor (src/client.js) or just try running it the way you are describing. I'm pretty confident it will crash. If it will not, please share entire fragment of a code you are using so that I understand your use-case better, because currently it looks wrong.

Reply:
Actually, I'm not seeing any typings for parseConnectionString function. I think that's where your problem is. It needs to be typed to return Config instance, then you could pass its results to the constructor of Client.

Reply:
I'm thinking I've touched the wrong line. What I want is to allow `knex(DATABASE_URL)` accept a string. Sorry should have added that context to the PR description.

Reply:
And now I see it seems to accept the following format: `knex({ connection: DATABASE_URL })`, which totally works for me.

Reply:
Ah, that's whole different story:
```
export default function Knex(config) {
  // If config is a string, try to parse it
  if (typeof config === 'string') {
    const parsedConfig = Object.assign(parseConnection(config), arguments[2]);
    return new Knex(parsedConfig);
  }
```

Reply:
Would be cool to expand typing, though, as passing just a string is also legit

Reply:
Right! Will try again.
(I've deleted the fork accidentally so will have to put up separate pr as I cannot reopen this one)

Reply:
https://github.com/tgriesser/knex/pull/3105